### ISSUE TITLE
Can't control RX-A3010

### ISSUE THREAD

**Issue by TMaxElectronics** [mvdwetering/yamaha_ynca#9]
The integration has a problem when using it with the nine channel RX-A3010 receiver.

I can connect to the receiver fine and all zones show up as entities, but trying to change anything about any entity I get this error:

> Logger: homeassistant.components.websocket_api.http.connection
> Source: custom_components/yamaha_ynca/media_player.py:218
> Integration: Home Assistant WebSocket API (documentation, issues)
> First occurred: July 13, 2022 at 2:13:23 AM (10 occurrences)
> Last logged: 1:12:08 PM
> 
> [140494764384624] '9ch Stereo' is not a valid SoundPrg
> [140494764167552] '9ch Stereo' is not a valid SoundPrg
> [140494820888480] '9ch Stereo' is not a valid SoundPrg
> Traceback (most recent call last):
>   File "/usr/src/homeassistant/homeassistant/components/websocket_api/commands.py", line 193, in handle_call_service
>     await hass.services.async_call(
>   File "/usr/src/homeassistant/homeassistant/core.py", line 1713, in async_call
>     task.result()
>   File "/usr/src/homeassistant/homeassistant/core.py", line 1750, in _execute_service
>     await cast(Callable[[ServiceCall], Awaitable[None]], handler.job.target)(
>   File "/usr/src/homeassistant/homeassistant/helpers/entity_component.py", line 204, in handle_service
>     await service.entity_service_call(
>   File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 649, in entity_service_call
>     entity.supported_features is None
>   File "/config/custom_components/yamaha_ynca/media_player.py", line 218, in supported_features
>     if self._zone.soundprg:
>   File "/usr/local/lib/python3.10/site-packages/ynca/zone.py", line 137, in soundprg
>     SoundPrg(self._attr_soundprg) if self._attr_soundprg is not None else None
>   File "/usr/local/lib/python3.10/enum.py", line 385, in __call__
>     return cls.__new__(cls, value)
>   File "/usr/local/lib/python3.10/enum.py", line 710, in __new__
>     raise ve_exc
> ValueError: '9ch Stereo' is not a valid SoundPrg

I'm not familiar enough with python to just add this myself (as I have no idea how to change a file that came from pip install), but I suspect all that is needed is to add

`   NINE_CH_STEREO = "9ch Stereo"` 

to [this line](https://github.com/mvdwetering/ynca/blob/bbfaef0324fa3c64572ad2915d170c5d43ce359b/ynca/constants.py#L23) in the ynca library

**Comment by mvdwetering** [comment]
Yeah, that should fix it. I just released a new version with that line added.

Unfortunately the YNCA spec I found online was quite old and did not have this mode in the list.

**Comment by TMaxElectronics** [comment]
Yeah that seems to have fixed it, but I do 7ch stero mode for my amp too :D 
Doesn't matter though, as it doesn't seem to care about a mode being selected that it doesn't recognise.

I guess the 9ch stereo option may well be part of the official protocol, but only used by like one amp because no others actually have 9 channels. 

Thanks for the very quick fix and for putting this integration out there :)

**Comment by mvdwetering** [comment]
FYI I looked a bit closer at the RX-A3000 soundmode list and there seem to be more mismatches between the list of soundmodes I have in `ynca` and the ones that your receiver supports, like "Hall in Amsterdam" and "Church in Freiburg". This is assuming the RX-A3010 supports at least the same list as RX-A3000. 

This means that if you select one of those unmatched modes you will run into the same issue again. I am looking into how to support the varying lists between model types as I can not ask the supported list from the receiver itself.

btw does your receiver really support 7ch-stereo mode? I just found the YNCA specs for the RX-A3020 and that one _does_ list 9ch-stereo, but not 7ch-stereo.

**Comment by TMaxElectronics** [comment]
Hmm interesting that some modes are still missing. Doesn't seem to bother HA as much as the 9ch stereo option not being there though. That didn't even let me switch the device on even when a different scene was selected. Selecting the unavailable 7ch stereo options doesn't seem to bother the amplifier either.

I just dumped the list of programms my RX-A3010 supports with its associated group name:

STRAIGHT
- Straight

CLASSICAL
- Hall in Munich
- Hall in Vienna
- Hall in Amsterdam
- Church in Freiburg
- Church in Royaumont
- Chamber

LIVE/CLUB
- Village Vanguard
- Warehouse Loft
- Cellar Club
- The Roxy Theatre
- The Bottom Line

ENTERTAINMENT
- Sports
- Action Game
- Roleplaying Game
- Music Video
- Recital/Opera

MOVIE
- Standard
- Spectacle
- Sci-Fi
- Adventure
- Drama
- Mono Movie

STEREO
- 2ch Stereo
- 9ch Stereo

SURROUND DECODE
- Surround Decoder

Hope that helps :)

**Comment by mvdwetering** [comment]
Thanks for the list, this matches the list that should be shown with the latest version of the integration.

I am now including lists for all known models and relaxed the handling of the soundmodes so it does not break anymore when there is an unknow soundmode read from the device.
